### DISCUSSION TITLE
Using select from NodOn SIN-4-FP-21 (or other)

### DISCUSSION THREAD

**Discussion by AthAshino** [jmcollin78/versatile_thermostat#613] [Ideas]
Hey!

I'm a new user of this integration and really love it! It is really complete and full of features. But there is something that "triggers" me. When I came with my NodOn SIN-4-FP-21 at first I was confused with it compatible directly with the integration. Then, reading the README, I understood that I must use a switch.
So I did it. But.. why?

My NodOn uses a selector with each mode for my radiator. Using the switch, I can only have 2: Eco and Comfort. Problem is, how can I manage boost or frost protection (or even real OFF).

Do you plan to add the possibility to create a VTherm using a selector so the VTherm directly uses it instead of tricking with a switch?
Is it at least possible? Or did I miss something that can explain that it is not. I would be happy to contribute on this features, but before putting my hands on it, I prefer to double check if there is any reason this does not exist yet.

Thank you! And thank you over all for this huge integration!

**Comment by jmcollin78** [comment]
Hello @AthAshino ,

You are right. But I cannot deal with all cases. You have find that this is a "huge integration". You are talking about a new VTherm type (over_switch, over_cliamte, ...) and this can be easily done by a virtual switch.

The other and maybe the main reason is that, if VTherm only select the preset of your Nodon, how will it regulate the temperature with an external temperature sensor ? Do simply select the correct preset (and forget temperature regulation) you don't need VTherm. A simple thermostat does the trick.

**Comment by AthAshino** [comment]
Thanks @jmcollin78 !
Oh ok I understand the reasons. Then I have another question. Why using the "eco" mode as the turn off function with the virtual switch? Why not using the real off state. Using eco mode will prevent frost protection to work if I'm not wrong.
I mean :
- If I put my VTherm in frost protection, it will make my heater off until the temperature is below 7°
- But turning off means putting the mode to "eco" which is "on until comfort -3° is reached which will not be 7°

Is it not preferable to make the switch putting the heater on "off" mode when the switch is turned off?

Thanks!

**Reply by jmcollin78** [reply]
> Then I have another question. Why using the "eco" mode as the turn off function with the virtual switch? Why not using the real off state

You can do that of course if you think it is more relevant in your case. This is just an example of a virtual switch

**Reply by AthAshino** [reply]
Why putting it to "eco" in the example? Is there any reason I don't see? (just for my knowledge)

**Reply by jmcollin78** [reply]
I'm not the author of the example. So I don't know. It should be a low value Anti-Frost, Eco, turning off looks like great possibilities
